fragalysis_api/xcimporter/align.py

```diff
@@ -98,138 +111,407 @@ def __get_length_and_resolution(self, file):
-
-    def __get_header(self, pdb_file):
-        """
-        Identifies the section of a PDB which contains the headers ATOM/HETATM
-        :param pdb_file: The pdb to acquire the header locations of
-        :return: front locations of the ATOM/HETATM headers in the given pdb, end locations of the ATOM/HETATM headers in the given pdb       
-        """
-        with open(pdb_file) as handle:
-            switch = 0
-            header_front, header_end = [], []
-
-            for line in handle:
-
-                if line.startswith('ATOM'): switch = 1
-
-                if line.startswith('HETATM'): switch = 2
-
-                if switch == 0: header_front.append(line)
-
-                if (switch == 2) and not line.startswith('HETATM'): header_end.append(line)
-
-        return header_front, header_end
-
-    def _save_align(self, name, pdb_base, out_dir):
-        """
-        Saves modified pdb as pdb file again. It also ensures the pdb header is kept in the new file.
-        :param name: name of pdb
-        :param pdb_base: coordinates of atom in the pdb format
-        :param out_dir: directory to save new pdb file in
-        :return: a saved pdb file
-        """
-        if not os.path.exists(out_dir):  # Creating output directory if it doesn't already exist
-            os.makedirs(out_dir)
-
-        pdb_file = self.__get_pdb_file(name)
-        header_front = self.__get_header(pdb_file)[0]
-
-        with open(os.path.join(out_dir, f'{name}_bound.pdb'), 'w') as handle:
-
-            new_pdb = header_front + [pdb_base]
-
-            for line in new_pdb:
-                handle.write(line)
+    def align_to_reference(self, in_file, reference, out_dir):
+        '''
+        Aligns a single pdb file to a reference and adds it to the specified output directory.
+        :param in_file: filepath to corresponding pdb file to align. Accompanying map files should be located within
+            the same directory as input file.
+        :param reference: the reference pdb file to align to.
+        :param out_dir: The desired output directory for the aligned pdb file.
+        :return: an aligned pdb file in the output directory with the same name as input.
+        '''
+        input_files = in_file
+        map_list = self._get_maplist
+        base_names = os.path.splitext(os.path.basename(input_files))[0]
+        crystals = [y for y in [x for x in [base_names] if 'event' not in x] if 'fofc' not in y]
+        ref = reference
+        dir = self.directory
+        mono = self.mono
+        for num, name in enumerate(crystals):
+            reference_pdb = Structure.from_file(file=Path(ref))
+            all_maps = [j for j in map_list if name in j]
+            current_pdb = Structure.from_file(file=Path(in_file))
+            try:
+                current_pdb, transform = current_pdb.align_to(other=reference_pdb, monomerized=mono)
+            except Exception as e:
+                # Poorly Documented. Use better stuff...
```

Comment:
For now - just add a note of what these exceptions are/when they happen if you can - assuming they're hard to understand because they're C++ errors?